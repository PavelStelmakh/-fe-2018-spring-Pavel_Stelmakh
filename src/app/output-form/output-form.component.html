<div>
  <div>
    <span>Name:</span>
    <span>{{formValue?.username}}</span>
  </div>
  <div>
      <span>Age:</span>
      <span>{{formValue?.userage}}</span>
  </div>
  <div>
      <span>Birthday:</span>
      <span>{{formValue?.birthday}}</span>
  </div>
  <div>
      <span>Date of login:</span>
      <span>{{formValue?.dateOfLogin}}</span>
  </div>
  <div>
      <span>Date of notification:</span>
      <span>{{formValue?.DateOfNotification}}</span>
  </div>
</div>
