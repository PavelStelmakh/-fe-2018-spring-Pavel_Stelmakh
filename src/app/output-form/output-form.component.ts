import { Component, OnInit } from '@angular/core';
import { UsersService } from '../users.service';
import { User } from '../../../models/User';
import * as moment  from 'moment';

@Component({
  selector: 'app-output-form',
  templateUrl: './output-form.component.html',
  styleUrls: ['./output-form.component.scss']
})
export class OutputFormComponent implements OnInit {
  formValue: User;

  constructor(private user: UsersService) { }

  ngOnInit() {
    this.user.subject.subscribe((value: User) => {
      this.formValue = value;
      if (value.id) {
        const dateOfNextNotif = moment(value.dateOfNextNotif).utcOffset('+04:00').format('DD-MMM-YY');
        this.formValue.dateOfBirth = moment(value.dateOfBirth).utcOffset('+04:00').format('YYYY/MM/DD');
        this.formValue.dateOfFirstLogin = moment(value.dateOfFirstLogin).utcOffset('+04:00').format('DD MMMM YYYY');
        this.formValue.dateOfNextNotif = dateOfNextNotif !== 'Invalid date' ? dateOfNextNotif: value.dateOfNextNotif;
      }
    });
  }

}
