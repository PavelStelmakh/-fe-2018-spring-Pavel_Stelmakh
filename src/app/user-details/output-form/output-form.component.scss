:host {
    color: #ccc;
    display: block;
    width: 450px;
    margin: 0 auto;
}

span {
    display: inline-block;
    font-size: 1.1rem;
    width: 225px;
}

div > div {
    margin-top: 10px;
}