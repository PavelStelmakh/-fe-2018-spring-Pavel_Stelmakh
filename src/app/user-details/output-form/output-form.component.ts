import { Component, OnInit } from '@angular/core';
import { User } from '../../../../models/User';
import * as moment  from 'moment';
import { Store, select } from '@ngrx/store';
import * as profileReducer from '../../reducers/profile.reduser';

@Component({
  selector: 'app-output-form',
  templateUrl: './output-form.component.html',
  styleUrls: ['./output-form.component.scss']
})
export class OutputFormComponent implements OnInit {
  formValue: User;

  constructor(private store: Store<profileReducer.State>) { }

  ngOnInit() {
    this.store.pipe(select(profileReducer.getProfileUser)).subscribe((value: User) => {
      this.formValue = value || {} as User;
      if (value) {
        const dateOfNextNotif = moment(value.dateOfNextNotif).utcOffset('+04:00').format('DD-MMM-YY');
        this.formValue.dateOfBirth = moment(value.dateOfBirth).utcOffset('+04:00').format('YYYY/MM/DD');
        this.formValue.dateOfFirstLogin = moment(value.dateOfFirstLogin).utcOffset('+04:00').format('DD MMMM YYYY');
        this.formValue.dateOfNextNotif = dateOfNextNotif !== 'Invalid date' ? dateOfNextNotif: value.dateOfNextNotif;
      }
    });
  }

}
