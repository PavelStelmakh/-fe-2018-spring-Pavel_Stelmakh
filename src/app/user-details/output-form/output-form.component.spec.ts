import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from "@angular/router/testing";
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { OutputFormComponent } from './output-form.component';
import { Observable, of } from 'rxjs';
import { TranslateModule, TranslateLoader, TranslateService } from '@ngx-translate/core';
import { Store } from '@ngrx/store';
import { hot } from 'jasmine-marbles';
import * as profileReducer from '../../reducers/profile.reduser';

const translations = {
  "user-details": {
    "output-form": {
      "name": "Name",
      "password": "Password",
      "age": "Age",
      "birthday": "Birthday",
      "dateOfLogin": "Date of login",
      "information": "Information",
      "dateOfNotification": "Date of notification"
    }
  }
};

class FakeLoader implements TranslateLoader {
  getTranslation(lang: string): Observable<{}> {
    return of(translations);
  }
}

describe('OutputFormComponent', () => {
  let fixture: ComponentFixture<OutputFormComponent>;
  let translate: TranslateService;
  let store: Store<profileReducer.State>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        TranslateModule.forRoot({
          loader: {provide: TranslateLoader, useClass: FakeLoader}
        })
      ],
      declarations: [
        OutputFormComponent
      ],
      providers: [
        {
          provide: Store,
          useValue: {
            dispatch: jest.fn(),
            pipe: jest.fn(() => hot('-a', { 
              a: {
                "id": 1,
                "name": "Admin",
                "age": "19",
                "password": "Admin",
                "dateOfBirth": "2011-10-05T00:00:00.000Z",
                "dateOfFirstLogin": "2011-10-05T14:48:00.000Z",
                "dateOfNextNotif": "2011-10-05T14:48:00.000Z",
                "information": "I'm ADMIN",
                "role": "admin"
              }
            }))
          }
        }
      ],
      schemas: [
       NO_ERRORS_SCHEMA
      ]
    }).compileComponents();

  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(OutputFormComponent);
    translate = TestBed.get(TranslateService);
    store = TestBed.get(Store);
  });

  it('renders markup to snapshot', () => {
    translate.use('en');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});
