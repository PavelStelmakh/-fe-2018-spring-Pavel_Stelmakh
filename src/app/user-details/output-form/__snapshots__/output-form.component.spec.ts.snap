// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputFormComponent renders markup to snapshot 1`] = `
<app-output-form
  store={[Function Object]}
>
  <div>
    <div>
      <span>
        Name:
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span>
        Age:
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span>
        Birthday:
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span>
        Date of login:
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span>
        Information:
      </span>
      <span>
        
      </span>
    </div>
    <div>
      <span>
        Date of notification:
      </span>
      <span>
        
      </span>
    </div>
  </div>
</app-output-form>
`;
